### PR TITLE
Port CMake Changes from jdk11

### PR DESCRIPTION
**Pull request by dnakamura**
ibmruntimes/openj9-openjdk-jdk11#185
ibmruntimes/openj9-openjdk-jdk11#161
ibmruntimes/openj9-openjdk-jdk11#153
ibmruntimes/openj9-openjdk-jdk11#68

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by DanHeidinga** [comment]
@keithc-ca Can you review this?

File: closed/autoconf/custom-spec.gmk.in [hunk 1]
```diff
@@ -118,3 +118,9 @@ OPENSSL_BUNDLE_LIB_PATH:=@OPENSSL_BUNDLE_LIB_PATH@
 OPENSSL_CFLAGS:=@OPENSSL_CFLAGS@
 OPENSSL_DIR:=@OPENSSL_DIR@
 WITH_OPENSSL:=@WITH_OPENSSL@
+
+ifeq (true,$(OPENJ9_ENABLE_CMAKE))
+  OPENJ9_VM_BUILD_DIR := $(OUTPUT_ROOT)/vm/runtime
+else
+  OPENJ9_VM_BUILD_DIR := $(OUTPUT_ROOT)/vm
+endif
```
Comment (keithc-ca) [inline closed/autoconf/custom-spec.gmk.in]: This doesn't look right: `OUTPUT_ROOT` is used in jdk8, but in jdk11+ it should be `OUTPUTDIR`.

File: closed/autoconf/custom-hook.m4 [hunk 2]
```diff
@@ -398,40 +398,42 @@ AC_DEFUN_ONCE([OPENJ9_THIRD_PARTY_REQUIREMENTS],
...
-    AC_MSG_NOTICE([Could not find freemarker.jar])
-    AC_MSG_ERROR([Cannot continue])
-  else
-    AC_MSG_RESULT([yes])
-    AC_MSG_CHECKING([checking that '$with_freemarker_jar' exists])
-    if test -f "$with_freemarker_jar" ; then
+  FREEMARKER_JAR=
+  if test "x$with_cmake" == x ; then
+    if test "x$with_freemarker_jar" == x ; then
+      AC_MSG_RESULT([no])
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4]: This should be more like the same file in jdk11; in particular `AC_MSG_CHECKING([that freemarker location is set])` should only be within the branch where cmake is not in use.

**Comment by dnakamura** [comment]
@keithc-ca Should be all fixed up now

**Comment by keithc-ca** [comment]
Please fix the signed-off-by line - it is missing the angle brackets (`<>`); it should be
```
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>
```

**Comment by dnakamura** [comment]
interesting, not sure how I managed that. Fixed

**Comment by keithc-ca** [comment]
Jenkins compile x86-64_linux,x86-64_linux_cm jdknext

**Comment by keithc-ca** [comment]
Local cmake build on x86-64_linux passed; eclipse/openj9#9051 will enable future pull-request builds.

**Pull request merged**